docs/source/client_gui/gobject_signals.rst

```diff
@@ -46,14 +79,57 @@ events.
    exit. Subscribers to this signal can use it as a chance to display a warning
    dialog and cancel the operation.
 
+.. py:function:: message-deleted(row_id)
+
+   This signal is emitted when the user deletes a message entry. Subscribers
+   to this signal can use it to conduct an action after the entry is deleted.
+   Signal emission stage set to run first.
+
+   :param str row_id: The row ID that was deleted.
+
+.. py:function:: message-sent(target_uid, target_email)
+
+   This signal is emitted when the user sends a message. Subscribers
+   to this signal can use it to conduct an action after the message is sent,
+   and the information saved to the database.
+   Signal emission stage set to run last.
+
+   :param str target_uid: Message uid that was sent.
+   :param str target_email: Email address associated with the sent message.
+
 .. py:function:: rpc-cache-clear()
 
    This signal is emitted to clear the RPC objects cached information.
    Subsequent invocations of RPC cache enabled methods will return fresh
    information from the server.
+   Signal emission stage set to run first.
 
 .. py:function:: server-connected()
 
    This signal is emitted when the client has connected to the King Phisher
    server. The default handler sets the initial campaign optionally prompting
    the user to select one if one has not already been selected.
+   Signal emission stage set to run last.
+
+.. py:function:: server-disconnected()
+
+   This signal is emitted when the client has disconnected from the King Phisher
+   server. The default handler sets the initial campaign optionally prompting
+   the user to select one if one has not already been selected.
+   Signal emission stage set to run first.
+
+.. py:function:: sftp-client-start()
+
+   This signal is emitted when the client starts sftp transfer from within
```

Comment:
"starts the sftp client" not transfer, we don't know when or if a transfer will actually be initiated.